v1.10 backports 2022-04-19

tklauser opened the pull request:
* ~#19308 -- pull skb data at the entrance of from-containter (@liuyuan10)~
   - Merge conflicts resolved according to
     https://github.com/cilium/cilium/pull/19308#issuecomment-1099573718
   - Dropped for now due to https://github.com/cilium/cilium/pull/19482#issuecomment-1105117347
 * #19458 -- jenkinsfiles: Increase VM boot timeout (@pchaigno)
 * #19491 -- pkg/bpf: add map name in error message for OpenParallel (@aanm)
 * #19478 -- docs: improve description for session affinity with KPR (@julianwiedmann)
 * #19489 -- LRP minor improvements (@aditighag)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 19458 19491 19478 19489; do contrib/backporting/set-labels.py $pr done 1.10; done
```

pchaigno committed:
jenkinsfiles: Increase VM boot timeout

[ upstream commit cfec27a217259e2002932864fe69e1df072319bc ]

This commit increases the VM boot timeout while decreasing the overall
timeout :mindblown:

We currently run the vagrant-ci-start.sh script with a 15m timeout and
retry twice if it fails. That takes up to 45m in total if all attempts
fail, as in frequently happening in CI right now. In particular, if the
script simply fails because it's taking on average more than 15m then
it is likely to fail all three times.

This commit instead increases the timeout from 15m to 25m and removes
the retries. The goal is obviously to succeed on the first try :p

Ideally, we would investigate why it is now taking longer to start the
VM. But this issue has been happening for a long time. And because of
the retries, we probably didn't even notice the increase at the
beginning: if it takes on average 15min, it might fail half the time and
the test might still succeed most of the time. That is, the retries
participate to hide the increase.

Signed-off-by: Paul Chaignon <paul@cilium.io>
Signed-off-by: Tobias Klauser <tobias@cilium.io>

julianwiedmann committed:
docs: improve description for session affinity with KPR

[ upstream commit b7002f593b44850af162e8f8f5f01a0dcc92e65b ]

Use the correct terminology ('session affinity' vs 'service affinity'),
and fix a typo.

Signed-off-by: Julian Wiedmann <jwi@isovalent.com>
Signed-off-by: Tobias Klauser <tobias@cilium.io>

pchaigno reviewed (commented):
My changes look good to me. Thanks!

aditighag committed:
pkg/redirectpolicy: Improve error logs

[ upstream commit 6c34c93dce924bc24072c23a154a62cf568b4d38 ]

Improve error logs thrown by port
validation logic so that user can
take necessary actions.

Signed-off-by: Aditi Ghag <aditi@cilium.io>
Signed-off-by: Tobias Klauser <tobias@cilium.io>

aditighag committed:
docs: Update section title to improve readability

[ upstream commit 79d53af7d5f0c3b91c17ff7cb5cc0b203d34f1d1 ]

Local redirect policy requires Kube-proxy replacement, and
the feature flag to be enabled. Rename the section that
outlines these steps so that users are less likely to miss
them.

Suggested-by: Raymond de Jong <raymond.dejong@isovalent.com>
Signed-off-by: Aditi Ghag <aditi@cilium.io>
Signed-off-by: Tobias Klauser <tobias@cilium.io>

aanm committed:
pkg/bpf: add map name in error message for OpenParallel

[ upstream commit 29c3ebdc0ace37e3068945942d35fe3e3786f4cd ]

Signed-off-by: André Martins <andre@cilium.io>
Signed-off-by: Tobias Klauser <tobias@cilium.io>
